File: model/src/main/kotlin/net/dungeonhub/model/server/DiscordServerModel.java
package net.dungeonhub.model.server;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import me.taubsie.dungeonhub.common.DungeonHubService;
import me.taubsie.dungeonhub.common.entity.model.CreationModel;
import me.taubsie.dungeonhub.common.entity.model.Model;
import me.taubsie.dungeonhub.common.entity.model.UpdateModel;

@AllArgsConstructor
@Getter
@Setter
public class DiscordServerModel implements Model, CreationModel, UpdateModel<DiscordServerModel> {
    private long id;

    public static DiscordServerModel fromJson(String json) {
        return DungeonHubService.getInstance()
                .getGson()
                .fromJson(json, DiscordServerModel.class);
    }

    @Override
    public DiscordServerModel apply(DiscordServerModel discordServerModel) {
        discordServerModel.setId(id);
        return discordServerModel;
    }
}

File: model/src/main/kotlin/net/dungeonhub/model/discord_server/DiscordServerModel.kt
package net.dungeonhub.model.discord_server

import net.dungeonhub.structure.model.CreationModel
import net.dungeonhub.structure.model.Model
import net.dungeonhub.structure.model.UpdateModel
import net.dungeonhub.service.MoshiService

class DiscordServerModel(
    val id: Long
) : Model, CreationModel, UpdateModel<DiscordServerModel> {
    companion object {
        fun fromJson(json: String): DiscordServerModel {
            return MoshiService.moshi.adapter(DiscordServerModel::class.java).fromJson(json)!!
        }
    }
}